### ISSUE TITLE
Overflow-menu: Content is not within a landmark element

### ISSUE THREAD

**Issue by gptt916** [carbon-design-system/carbon#7305]
## Environment

> Operating system

mac

> Browser

Chrome

> Automated testing tool and ruleset

![image](https://user-images.githubusercontent.com/20601623/99451930-eb4ebc80-28f0-11eb-9be7-b2f92d2b2156.png)

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

x.23.2

> What did you expect to happen?

Should pass accessibility

> What happened instead?

Failed on landmark rule

> What WCAG 2.1 checkpoint does the issue violate?

![image](https://user-images.githubusercontent.com/20601623/99452147-2ea92b00-28f1-11eb-8833-4616cda7966f.png)

## Steps to reproduce the issue

![image](https://user-images.githubusercontent.com/20601623/99452584-b5f69e80-28f1-11eb-850f-c45d785f126b.png)

1. Page location: https://dev.console.test.cloud.ibm.com/docs/openshift?topic=openshift-openshift_cloud_paks#oc_cloud_pak_ov
Click the Overflow menu button.
2. Run accessibility checker

## Additional information

The checker is failing on the `span` and the two `button` elements in this screenshot: 
![image](https://user-images.githubusercontent.com/20601623/99452286-58fae880-28f1-11eb-8c3a-81107f2306ec.png)

**Comment by jenschlot** [comment]
Any status on when this issue might be resolved?

**Comment by emyarod** [comment]
I'm unable to reproduce the issue with our latest floating menu code

![image](https://user-images.githubusercontent.com/8265238/108106563-bc60f780-7053-11eb-8d15-fdb815833985.png)

**Comment by gptt916** [comment]
What's the logic for the component determining where to insert the floating menu? It looks like for Docs it's incorrectly inserting it into <body> instead of where the button is actually rendered in the dom structure.

![image](https://user-images.githubusercontent.com/20601623/108225329-d78e3e80-7109-11eb-8092-e4a8ea0fe813.png)

In the above screenshot, the overflow menu button is within the landmark, however when clicking it, the floating menu is rendered at the bottom in <body>

**Comment by emyarod** [comment]
the floating menu is created using React portals, and if no target node is supplied then it's attached to `document.body` https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/internal/FloatingMenu.js#L455

**Comment by gptt916** [comment]
I see in the source there is this snippet for getTarget that is passed to FloatingMenu:

https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/OverflowMenu/OverflowMenu.js#L435

```js
  /**
   * @returns {Element} The DOM element where the floating menu is placed in.
   */
  _getTarget = () => {
    const { current: triggerEl } = this._triggerRef;
    return (
      (triggerEl && triggerEl.closest('[data-floating-menu-container]')) ||
      document.body
    );
  };
```

Does this mean I need to put `data-floating-menu-container` attribute on the containing body?

**Comment by tw15egan** [comment]
@gptt916 Yes, I believe adding `data-floating-menu-container` to the container should solve the issue

**Comment by gptt916** [comment]
Doesn't look this this is documented on the component page, could this be added somewhere?

**Comment by tw15egan** [comment]
@gptt916 it looks like we only got about halfway through updating the docs on our components, I'll take some time today to update the `OverflowMenu` docs page and be sure to include a section on `data-floating-menu-container`
